File: Bootcamp-suzano-backend-python/Desafio-de-codigo/calculo_salarial.py


def calcular_imposto(salario):
    aliquota = 0.00
    if (salario >= 0 and salario <= 1100):
        aliquota = 0.05
    elif (salario >= 1100.01 and salario <= 2500):
        aliquota = 0.10
    else :
        aliquota = 0.15
    return aliquota * salario

valor_salario = float(input())
valor_beneficio = float(input())

valor_imposto = calcular_imposto(valor_salario)
saida = valor_salario - valor_imposto + valor_beneficio
print(f'{saida:.2f}')




def calcular_saldo(transacoes):
    saldo = 0

    # TODO: Itere sobre cada transação na lista:
    for transacao in transacoes:
        # TODO: Adicione o valor da transação ao saldo
        saldo += transacao

    # TODO: Retorne o saldo formatado em moeda brasileira com duas casas decimais:
    return f"Saldo: R$ {saldo:.2f}"


entrada_usuario = input()

entrada_usuario = entrada_usuario.strip("[]").replace(" ", "")

transacoes = [float(valor) for valor in entrada_usuario.split(",")]

# TODO: Calcule o saldo com base nas transações informadas:
resultado = calcular_saldo(transacoes)

print(resultado)

File: Bootcamp-suzano-backend-python/Desafio-de-codigo/calculo_salarial.js
// Desafios JavaScript na DIO tem funções "gets" e "print" acessíveis globalmente;
// "gets" : lê UMA linha com dado(s) de entrada (inputs) do usuário;
// "print": imprime um texto de saída (output), pulando linha.

// lê os valores de Entrada:
const valorSalario = parseFloat(gets());
const valorBeneficio = parseFloat(gets());

// Calcule o imposto através da função "calcularImposto":
const valorImposto = calcularImposto(valorSalario);

// Calcula e imprime a Saída (com 2 casas decimais):
const saida = valorSalario - valorImposto + valorBeneficio;
print(saida.toFixed(2));

// Função útil para o calculo do imposto (baseado nas alíquotasa).
function calcularImposto(salario) {
    let aliqota;
    if (salario >= 0 && salario <=1100) {
        aliquota = 0.05;
    } else if (salario >= 1100.0 && salario <=2500) {
        aliquota = 0.10;
    } else aliquota = 0.15;
    return aliqota * salario
    
}
